Mathlib/Geometry/Manifold/IntegralCurve.lean

```diff
@@ -0,0 +1,317 @@
+  refine HasMFDerivAt.comp t (hγ (t * a) ht) ⟨(continuous_mul_right _).continuousAt, ?_⟩
+  simp only [mfld_simps, hasFDerivWithinAt_univ]
+  exact HasFDerivAt.mul_const' (hasFDerivAt_id _) _
+
+lemma isIntegralCurvOn_comp_mul_ne_zero {a : ℝ} (ha : a ≠ 0) :
+    IsIntegralCurveOn γ v s ↔ IsIntegralCurveOn (γ ∘ (· * a)) (a • v) { t | t * a ∈ s } := by
+  refine ⟨fun hγ ↦ hγ.comp_mul a, fun hγ ↦ ?_⟩
+  have := hγ.comp_mul a⁻¹
+  simp_rw [smul_smul, inv_mul_eq_div, div_self ha, one_smul, mem_setOf_eq, mul_assoc,
+    inv_mul_eq_div, div_self ha, mul_one, setOf_mem_eq] at this
+  convert this
+  ext t
+  rw [Function.comp_apply, Function.comp_apply, mul_assoc, inv_mul_eq_div, div_self ha, mul_one]
+
+lemma IsIntegralCurveAt.comp_mul_ne_zero (hγ : IsIntegralCurveAt γ v t₀) {a : ℝ} (ha : a ≠ 0) :
+    IsIntegralCurveAt (γ ∘ (· * a)) (a • v) (t₀ / a) := by
+  obtain ⟨ε, hε, h⟩ := isIntegralCurveAt_iff.mp hγ
+  rw [isIntegralCurveAt_iff]
+  refine ⟨ε / |a|, div_pos hε (abs_pos.mpr ha), ?_⟩
+  convert h.comp_mul a
+  ext t
+  rw [mem_setOf_eq, Metric.mem_ball, Metric.mem_ball, Real.dist_eq, Real.dist_eq,
+    lt_div_iff (abs_pos.mpr ha), ← abs_mul, sub_mul, div_mul_cancel _ ha]
+
+lemma isIntegralCurveAt_comp_mul_ne_zero {a : ℝ} (ha : a ≠ 0) :
+    IsIntegralCurveAt γ v t₀ ↔ IsIntegralCurveAt (γ ∘ (· * a)) (a • v) (t₀ / a) := by
+  refine ⟨fun hγ ↦ hγ.comp_mul_ne_zero ha, fun hγ ↦ ?_⟩
+  have := hγ.comp_mul_ne_zero (inv_ne_zero ha)
+  rw [smul_smul, inv_mul_eq_div, div_self ha, one_smul, ← div_mul_eq_div_div_swap,
+    inv_mul_eq_div, div_self ha, div_one, Function.comp.assoc] at this
+  convert this
+  ext t
+  simp [inv_mul_eq_div, div_self ha]
+
+lemma IsIntegralCurve.comp_mul (hγ : IsIntegralCurve γ v) (a : ℝ) :
+    IsIntegralCurve (γ ∘ (· * a)) (a • v) := by
+  rw [isIntegralCurve_iff_isIntegralCurveOn] at *
+  exact hγ.comp_mul _
+
+lemma isIntegralCurve_comp_mul_ne_zero {a : ℝ} (ha : a ≠ 0) :
+    IsIntegralCurve γ v ↔ IsIntegralCurve (γ ∘ (· * a)) (a • v) := by
+  refine ⟨fun hγ ↦ hγ.comp_mul _, fun hγ ↦ ?_⟩
+  have := hγ.comp_mul a⁻¹
+  rw [smul_smul, inv_mul_eq_div, div_self ha, one_smul] at this
+  convert this
+  ext t
+  rw [Function.comp_apply, Function.comp_apply, mul_assoc, inv_mul_eq_div, div_self ha, mul_one]
+
+/-- If the vector field `v` vanishes at `x₀`, then the constant curve at `x₀`
+  is a global integral curve of `v`. -/
+lemma isIntegralCurve_const {x : M} (h : v x = 0) : IsIntegralCurve (fun _ ↦ x) v := by
+  intro t
+  rw [h, ← ContinuousLinearMap.zero_apply (R₁ := ℝ) (R₂ := ℝ) (1 : ℝ),
+    ContinuousLinearMap.smulRight_one_one]
+  exact hasMFDerivAt_const ..
+
+end Scaling
+
+variable (t₀) {x₀ : M}
+
+/-- For any continuously differentiable vector field and any chosen non-boundary point `x₀` on the
+  manifold, there exists an integral curve `γ : ℝ → M` such that `γ t₀ = x₀` and the tangent vector
+  of `γ` at `t` coincides with the vector field at `γ t` for all `t` within an open interval around
+  `t₀`.-/
```

Comment:
IMHO this comment is no easier to read than the formal statement. I suggest:
```suggestion
/-- Existence of local integral curves for a C¹ vector field at interior points of a smooth
manifold. -/
```
since the formal statement is available if someone wants more precision.